File: tests/fixtures/polyglot/client/src/even.ts
/**
 * Case d (D4): cross-file call cycle, same shape as
 * tests/fixtures/typescript/src/even.ts (R5, bare named-import call).
 */
import { isOdd } from './odd';

export function isEven(n: number): boolean {
  return n === 0 ? true : isOdd(n - 1);
}

File: tests/fixtures/java/src/main/java/com/example/evenodd/Even.java
package com.example.evenodd;

import static com.example.evenodd.Odd.isOdd;

/**
 * Case d (D4): cross-file call cycle (mutual recursion) with Odd.java. The
 * static import makes the call site a bare identifier, so this resolves via
 * R5 (project-unique bare name), not R1/R2 — same reasoning as the
 * TypeScript/Python fixtures' even/odd cycle (R4 only consults import facts
 * for Rust in v1 — see ../../../../../../README.md).
 */
public class Even {
    public static boolean isEven(int n) {
        if (n == 0) {
            return true;
        }
        return isOdd(n - 1);
    }
}
